09_elements.html

```diff
@@ -289,4 +289,202 @@ <h2 id="element-location-strategies">Element Location Strategies</h2>
+
+    <li>If <var>element</var> is the document's root element,
+     that is <code>document.documentElement</code>:
+
+     <ol>
+      <li>If the computed value of the <code>background-color</code>
+       property is "transparent", run these substeps:
+
+       <ol>
+        <li>If <var>element</var> is an HTML <code>HTML</code>
+         element [[!html51]], and the computed value of the
+         <code>background-color</code> style property of the first
+         <code>BODY</code> element descendant of the element in
+         <a href=http://www.whatwg.org/specs/web-apps/current-work/multipage/infrastructure.html#tree-order>tree
+         order</a>, relative to that element, is also "transparent",
+         return false.
+
+        <li>Otherwise return false.
+       </ol>
+     </ol>
+
+    <li>If <var>element</var> is an <code>option</code>
+     or <code>optgroup</code> element,
+     and <var>element</var>'s parent node is a <code>select</code> element:
+
+     <ol>
+      <li>Apply the <a>element displayed</a> algorithm on <var>element</var>'s parent node.
+      <li>If the return value is false, abort these steps and return that value.
+     </ol>
+
+    <li>If <var>element</var> is a <code>map</code> element,
+     run these substeps:
+
+     <ol>
+      <li>Let <var>any images visible</var> be a boolean initially
+       set to false.
+
+      <li>For each <code>img</code> element, <var>image element</var>
+       in the document with a <code>name</code> attribute matching
+       the value of <var>element</var>'s <code>usemap</code>
+       attribute, run these substeps:
+
+       <ol>
+        <li>Run the <a>element displayed</a> algorithm on
+         <var>image element</var> and set <var>any images visible</var>
+         to <var>any images visible</var> bitwise OR its return value.
+       </ol>
+
+      <li>If <var>any images visible</var> is true, abort these
+       steps and return its value.
+    </ol>
+
+    <li>If <var>element</var> is an <code>area</code> element:
+
+     <ol>
+      <li>For each ancestral element <var>parent</var>,
+       in <a href=http://www.whatwg.org/specs/web-apps/current-work/multipage/infrastructure.html#tree-order>tree order</a>,
+       run the following substeps:
+
+      <ol>
+       <li>If <var>parent</var> is a <code>map</code> element,
+        apply the <a>element displayed</a> algorithm on it.
+
+       <li>If the return value is false, abort these steps and
+        return that value.
+
+       <li>Otherwise apply step 7.1 on <var>parent</var>.
+      </ol>
+     </ol>
+
+    <li>If <var>element</var> is a [[!DOM4]] <a
+     href=http://www.w3.org/TR/2012/WD-dom-20120105/#interface-text>
+     text node</a>, return true.
+
+    <li>If it has equal to or more than one direct descendant elements:
+
+     <ol>
+      <li>Let <var>visible children</var> be a boolean initially set to false.
+
+     <li>For each descendant element <var>child</var>,
```

Comment:
if "descendant elements" really means "direct descendant elements" here, please change to "direct descendant elements". If the text really means to say "descendant elements" (all of them), it would be helpful to emphasize that in the text (because "direct descendant elements" mentioned two sentences above may create confusion).